src/main/java/org/apache/sysml/runtime/instructions/cp/EvalBuiltinNaryCPInstruction.java

```diff
@@ -0,0 +1,63 @@
+package org.apache.sysml.runtime.instructions.cp;
+
+import org.apache.sysml.api.mlcontext.MLContextConversionUtil;
+import org.apache.sysml.parser.DataIdentifier;
+import org.apache.sysml.runtime.DMLRuntimeException;
+import org.apache.sysml.runtime.controlprogram.FunctionProgramBlock;
+import org.apache.sysml.runtime.controlprogram.LocalVariableMap;
+import org.apache.sysml.runtime.controlprogram.caching.MatrixObject;
+import org.apache.sysml.runtime.controlprogram.context.ExecutionContext;
+import org.apache.sysml.runtime.matrix.operators.Operator;
+
+import java.util.ArrayList;
+
+/**
+ * Eval built-in function instruction
+ * Note: it supports only single matrix[double] output
+ */
+public class EvalBuiltinNaryCPInstruction extends BuiltinNaryCPInstruction {
+
+    public EvalBuiltinNaryCPInstruction(Operator op, String opcode, String istr, CPOperand output, CPOperand... inputs) {
+        super(op, opcode, istr, output, inputs);
+    }
+
+    @Override
+    public void processInstruction(ExecutionContext ec) throws DMLRuntimeException {
+        //1. inside the new instruction, get the correspond function statement block
+        FunctionProgramBlock fpb = ec.getProgram().getFunctionProgramBlock(null, inputs[0].getName());
```

Comment:
You might want to check if the given name contains a "::" to recognize given namespaces - in this case you would need to split it and pass namespace and function name separately to getFunctionProgramBlock. Also note if you where not given a string but a function identifier at parser level, you need to capture both namespace and function from the function identifier.

Reply:
But it seems that the parser doesn't accept the function identifier as the function argument.